x-pack/plugins/ml/server/lib/register_settings.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup } from 'kibana/server';
+import { i18n } from '@kbn/i18n';
+import { schema } from '@kbn/config-schema';
+import { FILE_DATA_VISUALIZER_MAX_FILE_SIZE } from '../../common/constants/settings';
+import { MAX_FILE_SIZE } from '../../common/constants/file_datavisualizer';
+
+export function registerKibanaSettings(coreSetup: CoreSetup) {
+  coreSetup.uiSettings.register({
+    [FILE_DATA_VISUALIZER_MAX_FILE_SIZE]: {
+      name: i18n.translate('xpack.ml.maxFileSizeSettingsName', {
+        defaultMessage: 'File Data Visualizer maximum file upload size',
```

Comment:
In the docs we don't differentiate between the "Data Visualizer" and the "File Data Visualizer" (I think it's because they're both on the same "Data Visualizer" tab). IMO it's not necessary to have the "File" qualifier here either:
```suggestion
        defaultMessage: 'Data Visualizer maximum file upload size',
```